funcx_web_service/routes/funcx.py

```diff
@@ -91,10 +95,14 @@ def auth_and_launch(user_id, function_uuid, endpoints, input_data, app, token, s
                 'reason': f'Function UUID:{function_uuid} could not be resolved. {e}'}
 
     # Make sure the user is allowed to use the function on this endpoint
-    for ep in endpoints:
-        if not authorize_endpoint(user_id, ep, function_uuid, token):
-            return {'status': 'Failed',
-                    'reason': f'Unauthorized access to endpoint: {ep}'}
+    try:
+        for ep in endpoints:
+            if not authorize_endpoint(user_id, ep, function_uuid, token):
+                return {'status': 'Failed',
+                        'reason': f'Unauthorized access to endpoint: {ep}'}
+    except Exception as e:
```

Comment:
Let's also report the endpoint ID in the reason to disambiguate which endpoint is the culprit if multiple endpoints are requested.

Reply:
Currently the endpoint ID also ends up in `authorize_endpoint` exception messages, but I think it's harmless to repeat it and could be useful if an unexpected error occurs